lib/puppet/provider/yumrepo/inifile.rb

```diff
@@ -182,14 +182,27 @@ def self.repofiles
 
   # Build a virtual inifile by reading in numerous .repo files into a single
   # virtual file to ease manipulation.
+  # Will skip any inifile that is found to be currently managed as a file resource
+  # in the catalog.
   # @api private
   # @return [Puppet::Provider::Yumrepo::IniConfig::File] The virtual inifile representing
   #   multiple real files.
-  def self.virtual_inifile
+  def self.virtual_inifile(resources = nil)
+    current_yumrepo_inis = []
+    resources&.each do |key|
+      current_yumrepo_inis << key[1]
```

Comment:
Since `resources` is a Hash, you could extract the values directly:

```diff
-    current_yumrepo_inis = []
-    resources&.each do |key|
-      current_yumrepo_inis << key[1]
+    current_yumrepo_inis = resources.values
```